eospy/cleos.py

```diff
@@ -123,12 +123,20 @@ def get_producers(self, lower_bound='', limit=50, timeout=30) :
     #####
     # transactions
     #####
-    def push_transaction(self, transaction, keys, broadcast=True, compression='none', timeout=30) :
+    def push_transaction(self, transaction, keys, broadcast=True, compression='none', timeout=30,
+                         fully_offline=False, chain_id=None) :
         ''' parameter keys can be a list of WIF strings or EOSKey objects or a filename to key file'''
-        chain_info,lib_info = self.get_chain_lib_info()
-        trx = Transaction(transaction, chain_info, lib_info)
+        if fully_offline:
+            assert transaction.get('ref_block_num') and transaction.get('ref_block_prefix'), \
+                'ref_block_num or ref_block_prefix must be provided'
+            trx = Transaction(transaction, dict(), dict())

```

Comment:
Is it still a valid transaction if we pass an empty dict for `chain_info` and `lib_info`?
I see that the `Transaction` constructor treats `ref_block_num` and `ref_block_prefix` as optional properties, but I'm not sure how the encoded transaction differs.

Reply:
`Transaction` constructor requires `chain_info` and `lib_info` only to get `ref_block_num` & `ref_block_prefix`, accordingly, if they weren't provided. If they are already provided, then the first two are not used. That's the reason for the assertion, to make sure that the former two are provided in `transaction` dictionary. Otherwise, a proper transaction cannot be built (disregarding the exception).

Reply:
I see, thanks for the clarification